mergify_engine/queue/queue_pause.py

```diff
@@ -0,0 +1,124 @@
+# -*- encoding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+import dataclasses
+import datetime
+import typing
+
+import daiquiri
+import msgpack
+
+from mergify_engine import date
+from mergify_engine import utils
+from mergify_engine.context import Repository
+
+
+LOG = daiquiri.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class QueuePauseData:
+
+    repository: Repository
+
+    # Stored in redis
+    name: str = dataclasses.field(metadata={"description": "Queue name"})
+    reason: str = dataclasses.field(metadata={"description": "Pause message"})
+    user_app_name: str = dataclasses.field(
+        metadata={"description": "App responsible for pause"}
+    )
+    pause_date: datetime.datetime = dataclasses.field(default_factory=date.utcnow)
+    pause_state: bool = True
+
+    class Serialized(typing.TypedDict):
+        name: str
+        reason: str
+        user_app_name: str
+        pause_date: datetime.datetime
+        pause_state: bool
+
+    @classmethod
+    async def get(
+        cls, repository: Repository, queue_name: str
+    ) -> typing.Optional["QueuePauseData"]:
+
+        try:
+            repo_id = repository.repo["id"]
+            installation = repository.installation
+            redis_hash = f"merge-pause~{installation.owner_id}"
+            redis_key = f"{repo_id}~{queue_name}"
+
+            with utils.yaaredis_for_queue() as redis_queue:
+                queue_pause_data_raw = await redis_queue.hget(redis_hash, redis_key)
+
+                if queue_pause_data_raw:
+                    # NOTE(Syffe): timestamp parameter means that timestamp variables will be converted to
+                    # datetime (value 3=to_datetime()). Other values can be used: 1=to_float(), 2=to_unix_ns()
+                    queue_pause_data = msgpack.unpackb(
```

Comment:
any reason we use msgpack here?
i like it but I've seen json mostly used in other places